src/EditorFeatures/CSharpTest/Workspaces/WorkspaceTests.cs

```diff
@@ -783,6 +785,100 @@ public async Task TestDocumentEvents()
                                     shortEventTimeout.Seconds));
         }
 
+        [Fact]
+        public async Task TestSourceGeneratedDocumentEvents()
```

Comment:
➡️ This test intentionally has the same form as `TestDocumentEvents` in the same file. It will be simplified by consolidation of workspace events, but not otherwise changed.